internal/volume/oneinch.go

```diff
@@ -0,0 +1,128 @@
+package volume
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+)
+
+type oneInchVolumeTracker struct {
+	etherscanbaseUrl string
+	etherscanApiKey  string
+	ethplorerBaseUrl string
+	ethplorerApiKey  string
+	logger           *logrus.Logger
+}
+
+func NewOneInchVolumeTracker(etherscanApiKey, ethplorerApiKey string) IVolumeTracker {
+	return &oneInchVolumeTracker{
+		etherscanbaseUrl: "https://api.etherscan.io",
+		ethplorerBaseUrl: "https://api.ethplorer.io",
+		etherscanApiKey:  etherscanApiKey,
+		ethplorerApiKey:  ethplorerApiKey,
+		logger:           logrus.WithField("module", "oneInch_volume_tracker").Logger,
+	}
+}
+func (o *oneInchVolumeTracker) SafeClose(closer io.Closer) {
+	if err := closer.Close(); err != nil {
+		o.logger.Error(err)
+	}
+}
+
+// TODO: add from/to filter to etherscan request
+func (o *oneInchVolumeTracker) FetchVolume(from, to int64, affiliate string) (map[string]float64, error) {
+	// #TODO check api for from & to parameters
+	url := fmt.Sprintf("%s/v2/api?chainid=1&module=account&action=txlistinternal&address=%s&apikey=%s", o.etherscanbaseUrl, affiliate, o.etherscanApiKey)
```

Comment:
We do not have etherscan api key , neither of ethplorer , so let's forget about oneinch volume for now  , as long as we track THORChain, that will be ok